core/drivers/gpio/gpio.c

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Microchip
+ */
+
+#include <drivers/gpio.h>
+#include <libfdt.h>
+#include <stdio.h>
+#include <tee_api_defines.h>
+#include <tee_api_defines_extensions.h>
+#include <tee_api_types.h>
+#include <util.h>
+
+struct gpio *gpio_get_dt_two_cells(struct dt_driver_phandle_args *a,
```

Comment:
As this helper function allocates memory that caller maybe need to free, could you rename this function adding `_alloc_` wording inside, e.g. `gpio_alloc_and_get_dt_two_cells()`. 

To keep the label short, I suggest `gpio_dt_alloc_pin()`.

Reply:
Done